src/io/src/DataLoaderBridge.cpp

```diff
@@ -0,0 +1,216 @@
+  foundCloud = pt.get_child_optional( "data_path.path_to_cloud" );
+  if(foundCloud)
+  {
+    this->path_to_cloud = pt.get<std::string>("data_path.path_to_cloud");
+    std::string cloud_extension = pt.get<std::string>("data_path.cloud_extension");
+    getListFile(path_to_cloud, clouds, cloud_extension, isCloudFile);
+    std::sort(clouds.begin(), clouds.end());
+    data_size = clouds.size();
+    haveCloud = true;
+  }
+
+  boost::optional< const boost::property_tree::ptree& > foundRGB;
+  foundRGB = pt.get_child_optional( "data_path.path_to_rgb" );
+  if(foundRGB)
+  {
+    this->path_to_rgb = pt.get<std::string>("data_path.path_to_rgb");
+    std::string rgb_extension = pt.get<std::string>("data_path.rgb_extension");
+    getListFile(path_to_rgb, images, rgb_extension, isRGBFile);
+    std::sort(images.begin(), images.end());
+    data_size = images.size();
+    haveRGB = true;
+  }
+
+  boost::optional< const boost::property_tree::ptree& > foundDepth;
+  foundDepth = pt.get_child_optional( "data_path.path_to_depth" );
+  if(foundDepth)
+  {
+    this->path_to_depth = pt.get<std::string>("data_path.path_to_depth");
+    std::string depth_extension = pt.get<std::string>("data_path.depth_extension");
+    getListFile(path_to_depth, depths, depth_extension, isDepthFile);
+    std::sort(depths.begin(), depths.end());
+    data_size = depths.size();
+    haveDepth = true;
+  }
+
+  this->isLoop = pt.get<bool>("option.isLoop", true);
+
+  if(!checkConsistency()){
+    outError("Provided data is not consistent, all must be a file or size of each kind of data must be equal!");
+    _newData = false;
+  }
+}
+
+bool DataLoaderBridge::setData(uima::CAS &tcas, uint64_t ts){
+  if(!newData())
+    return false;
+
+  rs::SceneCas cas(tcas);
+
+  if(haveCloud){
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_ptr (new pcl::PointCloud<pcl::PointXYZRGBA>);
+
+    std::string path;
+    if(!isCloudFile){
+      path = clouds[iterator];
+    }
+    else{
+      path = path_to_cloud;
+    }
+
+    if(pcl::io::loadPCDFile (path, *cloud_ptr) == -1){
+      outError("Could not load point cloud file as PCD type. Check path again!");
+    }
+
+    cas.set(VIEW_CLOUD, *cloud_ptr);
+  }
+
+  if(haveRGB){
+    std::string path;
+    if(!isRGBFile){
+      path = images[iterator];
+    }
+    else{
+      path = path_to_rgb;
+    }
+
+    this->color = cv::imread(path);
+    cv::resize(color, color, cv::Size(640, 480), 0, 0, cv::INTER_NEAREST);
+    cas.set(VIEW_COLOR_IMAGE, color);
+  }
+
+  if(haveDepth){
+    std::string path;
+    if(!isDepthFile){
+      path = depths[iterator];
+    }
+    else{
+      path = path_to_depth;
+    }
+
+    this->depth = cv::imread(path,  CV_LOAD_IMAGE_ANYDEPTH);
+    cv::resize(depth, depth, cv::Size(640, 480), 0, 0, cv::INTER_NEAREST);
+
+    if (depth.type() == CV_16UC1)
+      depth.convertTo(depth, CV_16UC1, 3000./65535, 0);
+    else
+      depth.convertTo(depth, CV_16UC1, 3000./255, 0);
```

Comment:
Have in mind that's magic numbers which worked for me: AFAIK there is no universal depth conversion from 8bit image, and 16bit images may be represented in different units (usually in mm.) so some kind of multiplier like `depth_scale_factor` in config file might be handy